cabal-install/Distribution/Client/Targets.hs

```diff
@@ -455,6 +455,14 @@ readPackageTarget verbosity = traverse modifyLocation
         error "TODO: readPackageTarget RepoTarballPackage"
         -- For repo tarballs this info should be obtained from the index.
 
+      RemoteSourceRepoPackage _srcRepo _ ->
+        error "TODO: readPackageTarget RemoteSourceRepoPackage"
+        -- This can't happen, because it would have errored out already
```

Comment:
The answer turns out to be pretty unsatisfying. "It can't happen because something else would break first."

Reply:
fair enough... that's still something useful to know :-)